torchvision/datasets/flickr.py

```diff
@@ -0,0 +1,152 @@
+from collections import defaultdict
+from PIL import Image
+from six.moves import html_parser
+
+import glob
+import os
+import torch.utils.data as data
+
+
+class Flickr8kParser(html_parser.HTMLParser):
+    """Parser for extracting captions from the Flickr8k dataset web page."""
+
+    def __init__(self):
+        super(Flickr8kParser, self).__init__()
+
+        # Data structure to store captions
+        self.annotations = {}
+
+        # State variables
+        self.in_table = False
+        self.current_tag = None
+        self.current_img = None
+
+    def handle_starttag(self, tag, attrs):
+        self.current_tag = tag
+
+        if tag == 'table':
+            self.in_table = True
+
+    def handle_endtag(self, tag):
+        self.current_tag = None
+
+        if tag == 'table':
+            self.in_table = False
+
+    def handle_data(self, data):
+        if self.in_table:
+            if data == 'Image Not Found':
+                self.current_img = None
+            elif self.current_tag == 'a':
+                img_id = data.split('/')[-2]
+                self.current_img = img_id
+                self.annotations[img_id] = []
+            elif self.current_tag == 'li' and self.current_img:
+                img_id = self.current_img
+                self.annotations[img_id].append(data.strip())
+
+
+class Flickr8k(data.Dataset):
+    """`Flickr8k Entities <http://nlp.cs.illinois.edu/HockenmaierGroup/8k-pictures.html>`_ Dataset.
+
+    Args:
+        root (string): Root directory where images are downloaded to.
+        annFile (string): Path to annotation file.
```

Comment:
nit: can we use `camel_case`?

Reply:
I agree that `lowercase_with_underscores` is preferred over `CamelCase` according to PEP 8. I chose `annFile` to be consistent with `CocoCaptions` and `CocoDetection`. Do you want me to change COCO as well? I worry that this will break backwards compatibility for people who explicitly specified:
```python
CocoCaptions(root=..., annFile=...)
```

Reply:
Let's not change COCO, but only this one.
We don't enforce / guarantee that the different datasets have the same constructor, so this should be fine.